### DISCUSSION TITLE
AWS Elastic Container Service installation ("Fargate")

### DISCUSSION THREAD

**Discussion by shrike71** [directus/directus#2866] [Archive]
Hi All,

Wondering if anybody is able to shed some light on this. We've been able to quite easily run the initial setup and implementation of the Directus 8 container service in an ECS Target Group, using the ReadOnly appliance setup that ECS provides as a super fast scalable container aka "Fargate" and environment variables. However, as a last step, we need to run this, as specified in the directus container installation documentation:

`docker-compose run --rm directus install --email email@example.com --password d1r3ctu5`

The service based (daemon), read-only nature of a Fargate container means that being able to run a "docker exec" or a "docker run" or "docker compose" as orchestration is not possible. Is there a gui or other equivalent to the above?

**Comment by ashx404** [comment]
This is what I do, 
Setup directus locally, connect it to the SQL instance (whatever database you are using, just add env variables to config file) and then just run this command. It creates a database there.

**Reply by shrike71** [reply]
Sorry ashx404,

We've got all that set up and working. Database is an AWS RDS instance of MySQL and all env variables are injected via AWS Secrets on container start. The issue is with trying to run the command above, with NO ACCESS to the executive layer of the running container (which fargate predicates). We need some sort of remote or WEB UI equivalent.

**Comment by shrike71** [comment]
Hi all,

We've got to the Directus CMS login screen using a strict AWS Fargate, Secrets and env var setup. We can also confirm that all env vars seem to be read and created via the task. Now is there a way we might be able to create the default admin user in this command: 

`docker-compose run --rm directus install --email email@example.com --password d1r3ctu5`

via a set of **direct INSERT statements into the RDS DB instead?** We have on-prem access to it via MySQL workbench

**Comment by jbarneyally** [comment]
when your service is up and running in ECS, select a running task, "Run more like this", configure the task as needed, and change the container override input (under advanced) to simply "install" or "install --email email@example.com --password d1r3ctu5"

by default, "directus" is the entrypoint of the container -- subsequent inputs are taken as sub commands

**Comment by sevetseh28** [comment]
@shrike71 could you share how you set it up in ECS? thanks!
